### PR TITLE
UIP-1447 extending JsonMessageParser

### PR DESCRIPTION
**Pull request by sabinaunified**
@wiyu

### CODE REVIEW

File: src/test/java/com/pinterest/secor/parser/JsonPathTimestampedParserTest.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        JsonPathTimestampedParser jsonPathParser = new JsonPathTimestampedParser(mConfig);
+
+        String resultSeconds[] = jsonPathParser.extractPartitions(mMessageWithSecondsTimestamp);
+        assertEquals(4, resultSeconds.length);
+        assertEquals("gender=female", resultSeconds[0]);
+        assertEquals("id=0", resultSeconds[1]);
+        assertEquals("dt=2014-07-21",resultSeconds[2]);
+        assertEquals("hr=02", resultSeconds[3]);
+    }
+
```
Comment (wiyu) [inline src/test/java/com/pinterest/secor/parser/JsonPathTimestampedParserTest.java]: Can you add test case were the `prefixToJsonPathMap` is empty? Just to make sure it works as timestamp partitioner.

File: src/main/config/unified/secor.unified.tags.properties [hunk 2]
```diff
@@ -6,14 +6,15 @@ secor.kafka.topic_blacklist=
 secor.kafka.group=secor_collector_tags
 
 # Parser class that extracts s3 partitions from consumed messages.
-secor.message.parser.class=com.pinterest.secor.parser.JsonPathParser
+secor.message.parser.class=com.pinterest.secor.parser.JsonPathTimestampedParser
 
 # S3 path where files are stored.
 secor.s3.path=tag
 
 # TODO: Need to incorporate the timestamp partition logic from "JsonMessageParser"
```
Comment (wiyu) [inline src/main/config/unified/secor.unified.tags.properties]: can you remove this comment now that you've added this? Also, make a comment here to explain the expected partitioned folder structure this config will produce.
